lib/types/src/memory_view.rs

```diff
@@ -64,6 +67,44 @@ where
             _phantom: PhantomData,
         }
     }
+
+    /// Creates a subarray view from this MemoryView.
+    pub fn subarray(&self, range: impl RangeBounds<usize>) -> Self {
+        let start: usize = match range.start_bound() {
+            Bound::Unbounded => 0,
+            Bound::Included(start) => *start,
+            Bound::Excluded(start) => *start + 1,
+        };
+        let end: usize = match range.end_bound() {
+            Bound::Unbounded => self.length,
+            Bound::Included(end) => *end,
+            Bound::Excluded(end) => *end - 1,
+        };
+        Self {
+            ptr: unsafe { self.ptr.add(start) },
+            length: (end - start),
+            _phantom: PhantomData,
+        }
+    }
```

Comment:
We must ensure it doesn't overflow outside the memory bounds.

Reply:
Fixed!